### ISSUE TITLE
function-token-no-space complaining about valid syntax

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#209]
The following declaration is causing an error:

```css
padding-top: calc(0.5 * (var(--input-size-height) - var(--lineheight-pxbase))); /* 9 */
```

It should not, because there is in fact no space between any of the function tokens and their parentheses.

**Comment by davidtheclark** [comment]
The reason for this is that `(var(` part, where there is a parenthesis used to group & establish order rather than to start a function.

Off the top of my head I can't think of a reliable way to differentiate between parentheses opening function arguments and parentheses that are grouping, like this one. Can anyone else?

If we can't, that probably puts a bullet in this rule, which was questionable anyway (as it is not valid CSS to have a space between token and parenthesis anyway, I believe).

**Comment by jeddy3** [comment]
> If we can't, that probably puts a bullet in this rule, which was questionable anyway

SGTM. Shall we remove it?

**Comment by davidtheclark** [comment]
@MoOx any big ideas? If not, I guess let's remove it.

**Comment by MoOx** [comment]
I am not sure why this rule inspect the body of the function itself. Maybe I am missing something here.
We might detect a start when you get something like `[a-zA-Z]\s(` (function) or no chars (`(+|-|*|/)\s(`).
But it's that too hard, we should drop it for now :/

**Comment by davidtheclark** [comment]
It is inspecting the body of the function to look for nested functions (like the `var()` functions nested in `calc()` in the problem code above).

I guess that because I realized you can have parentheses without functions I wonder if we can accurately determine whether a separated word is a function name or just some other keyword. Could we ever have a value that does this: `... pink (...`? Or would this ever happen: `... 10px (...`? In both those cases, we'd be wrong to assume that the space is unintentional.

**Comment by davidtheclark** [comment]
Ok, if nobody has any other ideas on how to handle this, let's just drop it and see if the need ever arises again.
